dxr/plugins/js/indexers.py

```diff
@@ -72,31 +53,52 @@ def __init__(self, path, contents, plugin_name, tree):
         self.analysis_path = join(join(join(tree.temp_folder, 'plugins/js'),
                                        relpath(dirname(self.absolute_path()), tree.source_folder)),
                                   basename(path) + '.data')
-        lines = []
+        # All lines from the analysis output file.
+        self.lines = []
+        # Map of line number -> byte offset to use for emitting refs.
+        self.offsets = []
         if self.is_interesting():
             with open(self.analysis_path) as analysis:
-                lines = self.parse_analysis(analysis.readlines())
-            lines = sorted(lines, key=lambda x: x['loc'])
-        self.analyzer = ReadAnalysis(tree, lines, contents)
+                self.lines = sorted((self.parse_analysis(line) for line in analysis), key=lambda x: x.loc)
```

Comment:
They're laid down in AST traversal order, which isn't necessarily the order of appearance in the source file.